src/ReplicatedStorage/Client/Tool/Gun/init.lua

```diff
@@ -85,19 +86,35 @@ function Gun.DrawShot(startPoint: Vector3, endPoint: Vector3, config: table)
 	beam.Anchored = true
 	beam.Locked = true
 	beam.CanCollide = false
+	beam.CanQuery = false
+	beam.CanTouch = false
 	beam.Size = Vector3.new(thickness, thickness, distance)
-	beam.CFrame = CFrame.new(startPoint, endPoint) * CFrame.new(0, 0, -distance / 2)
 
 	local mesh = Instance.new("BlockMesh")
 	mesh.Parent = beam
-	beam.Parent = bulletContainer
 
 	local travelSpeed = (distance/300) / (config.BulletDecoration.BulletSpeed or 1)
 	Debris:AddItem(beam, travelSpeed)
 
 	task.delay(1/30, function()
 		tweenOnce(mesh, TweenInfo.new(travelSpeed - 1/30, Enum.EasingStyle.Quad, Enum.EasingDirection.In), {Scale = Vector3.new(1, 1, 0), Offset = Vector3.new(0, 0, -distance/2)})
 	end)
+
+	-- BeamUpdater updates the beam's cframe for 4 frames so it doesn't look weird when the character moves a lot
+	local renderId = "BeamUpdater-"..os.clock()
+	local endTime = os.clock() + 4*(1/60)
```

Comment:
This code assumes the user's framerate is 60fps. Either the number and comment should be switched to a duration in seconds, or a frame counter should be used.

Reply:
I would suggest just using the time 0.1 seconds

Reply:
@4812571 This is sort of what's being done I think. They only half assume that the player's frame rate is 60 FPS, they calculate an end time based on that. They are effectively using the duration `0.066...` (`4/60`).